src/log-sync.ts

```diff
@@ -425,7 +437,13 @@ class LogSync implements LogSeverityFunctions {
           entry.resource ||
           this.logging.detectedResource;
         entry[LABELS_KEY] = options.labels || entry[LABELS_KEY];
-        this.transport.write(JSON.stringify(entry) + '\n');
+        if (this.options.useWriteRequest) {
+          const reqOpts = {
+            logName: this.formattedName_,
+            entries: arrify(entry) as StructuredJson[],
+          };
+          this.transport.write(JSON.stringify(reqOpts) + '\n');
+        } else this.transport.write(JSON.stringify(entry) + '\n');
       }
     } catch (err) {
       // Ignore errors (client libraries do not panic).
```

Comment:
Can we modify this comment to a more generic statement for errors and move it above line#448?

Reply:
Sure thing - any suggestions how to make it more generic?